src/Microsoft.Azure.SignalR.Protocols/ServiceProtocolConstants.cs

```diff
@@ -15,6 +15,7 @@ public static class ServiceProtocolConstants
         public const int UserDataMessageType = 8;
         public const int MultiUserDataMessageType = 9;
         public const int BroadcastDataMessageType = 10;
+        public const int BroadcastDataMessageWithMessageIdType = 100010; // 1xxxxx mean the message with ID, otherwise without ID

```

Comment:
Should we define different types for with message id? Or just add an optional field for message id?

Reply:
I just add a optional field for message ID in `MulticastDataMessage`, so that we only need to add a new message type in `ServiceProtocolConstants`. 

If add a new type for adding a field for message ID, then all messages that need the new message class...

Or we change the message type, since diffent message type share some common fileds, we can make every digit meaningful, one digit to one field, so we know how to write and parse the message, then it might be easier to extend message in the future.